### PR TITLE
adding backend setup

### PR DESCRIPTION
**Pull request by stringtheory1234**
Adding backend files general setup

### CODE REVIEW

File: backend/src/test/java/com/google/backend/DemoApplicationTests.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class DemoApplicationTests {
+
+	@Test
+	void contextLoads() {
+	}
```
Comment (priytanvani) [inline backend/src/test/java/com/google/backend/DemoApplicationTests.java:11]: Do we need this file?

File: backend/pom.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>2.3.0.RELEASE</version>
+		<relativePath/> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>com.google</groupId>
```
Comment (priytanvani) [inline backend/pom.xml:11]: Let's rename this as "com.google.gpay.virtualqueue"

File: backend/pom.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	<artifactId>backend</artifactId>
```
Comment (priytanvani) [inline backend/pom.xml:12]: Let's rename this as "virtualqueueservice"

File: backend/pom.xml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	<version>0.0.1-SNAPSHOT</version>
+	<name>demo</name>
```
Comment (priytanvani) [inline backend/pom.xml:14]: Change this?

File: backend/pom.xml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	<description>Demo project for Spring Boot</description>
```
Comment (priytanvani) [inline backend/pom.xml:15]: Change this?

File: virtualqueueservice/.mvn/wrapper/MavenWrapperDownloader.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import java.net.*;
+import java.io.*;
+import java.nio.channels.*;
+import java.util.Properties;
+
+public class MavenWrapperDownloader {
```
Comment (priytanvani) [inline virtualqueueservice/.mvn/wrapper/MavenWrapperDownloader.java:21]: For my knowledge, what is the use of this file?

File: virtualqueueservice/mvnw [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+# Optional ENV vars
+# -----------------
+#   M2_HOME - location of maven2's installed home dir
+#   MAVEN_OPTS - parameters passed to the Java VM when running Maven
+#     e.g. to debug Maven itself, use
+#       set MAVEN_OPTS=-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=8000
+#   MAVEN_SKIP_RC - flag to disable loading of mavenrc files
+# ----------------------------------------------------------------------------
+
+if [ -z "$MAVEN_SKIP_RC" ] ; then
```
Comment (priytanvani) [inline virtualqueueservice/mvnw:37]: Do we need to checkin this file?

File: virtualqueueservice/mvnw.cmd [hunk 8]
```diff
@@ -0,0 +1,182 @@
+@REM ----------------------------------------------------------------------------
```
Comment (priytanvani) [inline virtualqueueservice/mvnw.cmd:1]: Do we need to check in this file?

File: virtualqueueservice/.mvn/wrapper/MavenWrapperDownloader.java (hunk 6 above)
Comment (stringtheory1234) [reply virtualqueueservice/.mvn/wrapper/MavenWrapperDownloader.java:21]: I guess that maven is downloaded using this file. It is necessary for executing the project.

**Pull request merged**